### PR TITLE
Render the cortellis api document for public user and Skus null (snaoshot)

### PR DESCRIPTION
**Pull request by darshan-clvt**
### For Public user we have enabled the cortelllis api doc 

1. only api doc 
2. no swagger button no tryout column in application
3. not skus descriptions also.
### code changes 

1. with out customID skus wont be fetched when customId is not found it will throw error 
2. we check for customId undefined 
3. we are handling null exception in for getting skus data
### ### Enabling to tryout button for apis related cortellis UI in cortellis-api-collection

1. add in  async.parallel to get the subscriptions of cortellis-api-collection
2. pass the results while rendering the UI 
3. updateing apikey when auth key-auth for url of swagger doc button
4. when apikey in none , no apikey is passed

### CODE REVIEW

File: src/ui/routes/apis.js [hunk 1]
```diff
@@ -327,8 +327,32 @@ router.get('/:api', function (req, res, next) {
...
-                const trueid = customId.split(":")
-                const sanitizedId = trueid.length > 1 ? trueid[1] : null;
+                let trueid, sanitizedId;
+                if (customId){
+                    trueid = customId.split(":")
+                    sanitizedId = trueid.length > 1 ? trueid[1] : null;
+                }
+                if ( customId === undefined || trueid === undefined  || sanitizedId === null ) {
+                    // Render the page directly without making the API request
+                    debug("I AM IF I AM GOING FIRST")
```
Comment (saichrn7) [inline src/ui/routes/apis.js]: remove this test dubug statement

**Review by saichrn7 — CHANGES_REQUESTED** [review]

File: src/ui/routes/apis.js [hunk 2]
```diff
@@ -183,12 +183,27 @@ router.get('/:api', function (req, res, next) {
             for (let i = 0; i < userInfo.applications.length; ++i)
                 appIds.push(userInfo.applications[i].id);
         }
-
+        let cortelliesapiSwap = [];
```
Comment (saichrn7) [inline src/ui/routes/apis.js]: rename variable names to meaningful one..all has same swap word.

File: src/ui/routes/apis.js [hunk 3]
```diff
@@ -183,12 +183,27 @@ router.get('/:api', function (req, res, next) {
...
+                if (api.hasOwnProperty(key)) {
+                    cortelliesapiSwap.push(api[key]);
+                }}
+        });
         // Note: callback and results are used all the time, but in the end, all's
         // good, as the variable scopes are playing nice with us. Just watch out.
         async.parallel({
             getSubs: function (callback) {
                 async.map(appIds, function (appId, callback) {
+                    for (let i = 0; i < cortelliesapiSwap.length; i++) {
```
Comment (saichrn7) [inline src/ui/routes/apis.js]: intead of iterationg, can you check if array includes method works ?

File: src/ui/routes/apis.js [hunk 4]
```diff
@@ -312,8 +327,40 @@ router.get('/:api', function (req, res, next) {
...
             apiInfo.hasProtectedAuthMethods = hasProtectedMethods;
             apiInfo.hasSwaggerApplication = hasSwaggerApplication;
+
             // See also views/models/api.json for how this looks
-            
+            let cortelliesapiIdValues = [];
+            let CortellisBundleApikey = [];
+            //check extracting apiId in json file in content
+            // let cortellisMainApi = req.app.portalGlobals.cortellisUi.cortelliesApi.cortelliesMainApiId;
+            let cortelliespiIds = req.app.portalGlobals.cortellisUi.cortelliesApi.InvestigationalApi;
```
Comment (saichrn7) [inline src/ui/routes/apis.js]: follow camelCase for all variable names

File: src/ui/routes/apis.js [hunk 5]
```diff
@@ -312,8 +327,40 @@ router.get('/:api', function (req, res, next) {
...
+                cortelliespiIds.forEach(function (api) {
+                    for (let key in api) {
+                        if (api.hasOwnProperty(key)) {
+                            cortelliesapiIdValues.push(api[key]);
+                        }}
+                });
+            //check for only api ids json file in content
+            let matchingValue = null;
```
Comment (saichrn7) [inline src/ui/routes/apis.js]: same thing here, please write the in common place and reuse it ?

**Review by saichrn7 — CHANGES_REQUESTED** [review]

File: src/ui/routes/apis.js (hunk 3 above)
Comment (darshan-clvt) [reply src/ui/routes/apis.js]: no it will not work Sai i have checked. it will simply pass true for other apis

File: src/ui/routes/apis.js (hunk 4 above)
Comment (daru19k) [reply src/ui/routes/apis.js]: sure

**Pull request merged**